tools/rna_tools/rnaformer/infer_rnaformer.xml

```diff
@@ -0,0 +1,198 @@
+                new_module.weight.copy_(target.weight)
+                if target.bias is not None:
+                    new_module.bias.copy_(target.bias)
+
+            elif isinstance(target, torch.nn.Conv2d):
+                kernel_size = target.kernel_size[0]
+                new_module = lora.Conv2d(target.in_channels, target.out_channels, kernel_size,
+                                        padding=(kernel_size - 1) // 2, bias=target.bias is not None,
+                                        **lora_config)
+
+                new_module.conv.weight.copy_(target.weight)
+                if target.bias is not None:
+                    new_module.conv.bias.copy_(target.bias)
+            setattr(parent, target_name, new_module)
+
+state_dict = torch.load(state_dict_file, map_location=torch.device('cpu'))
+
+model.load_state_dict(state_dict, strict=True)
+model_name = state_dict_file.split(".pth")[0]
+
+if torch.cuda.is_available():
+        model = model.cuda()
+
+        # check GPU can do bf16
+        if torch.cuda.is_bf16_supported():
+            model = model.bfloat16()
+        else:
+            model = model.half()
+
+model.eval()
+predicted_structures = []
+
+for sequence in sequences:
+    print(f"Sequence: {sequence}")
+    with torch.no_grad():
+        device = "cpu"
+
+        seq_vocab = ['A', 'C', 'G', 'U', 'N']
+        seq_stoi = dict(zip(seq_vocab, range(len(seq_vocab))))
+
+        pdb_sample = 1
+
+        length = len(sequence)
+        src_seq = torch.LongTensor(list(map(seq_stoi.get, sequence)))
+
+        sample = {}
+        sample['src_seq'] = src_seq.clone()
+        sample['length'] = torch.LongTensor([length])[0]
+        sample['pdb_sample'] = torch.LongTensor([pdb_sample])[0]
+
+        sequence = sample['src_seq'].unsqueeze(0).to(device)
+        src_len = torch.LongTensor([sequence.shape[-1]]).to(device)
+        pdb_sample = torch.FloatTensor([[1]]).to(device)
+
+        logits, pair_mask = model(sequence, src_len, pdb_sample)
+
+        pred_mat = torch.sigmoid(logits[0, :, :, -1]) > 0.5
+
+        pos_id = torch.where(pred_mat == True)
+        pos1_id = pos_id[0].cpu().tolist()
+        pos2_id = pos_id[1].cpu().tolist()
+        predicted_structure = f"Pairing index 1: {pos1_id} \nPairing index 2: {pos2_id}"
+        print(predicted_structure)
+
+        seqlen = len(sample['src_seq'])
+        dot_bracket =['.'] * seqlen
+        for i in range(len(pos1_id)):
+            open_index = pos1_id[i]
+            close_index = pos2_id[i]
+
+            if 0 <= open_index < len(dot_bracket) and 0 <= close_index < len(dot_bracket):
+                if dot_bracket[open_index] == '.' and dot_bracket[close_index] == '.':
+                    dot_bracket[open_index] = '('
+                    dot_bracket[close_index] = ')'
+        dot_bracket_str_pred = ''.join(dot_bracket)
+
+        print(f"Predicted Structure: {dot_bracket_str_pred}")
```

Comment:
```suggestion
```

Reply:
I can remove these two print statements but then do you suggest only including the list of the base pair positions in the output (without the original sequence and dot-bracket notation)? The output is now just a txt file of these redirected prints.